File: java/universalChat/src/main/java/com/github/fernthedev/packets/handshake/InitialHandshakePacket.java
package com.github.fernthedev.packets.handshake;

import com.github.fernthedev.packets.Packet;
import com.github.fernthedev.universal.encryption.util.RSAEncryptionUtil;

import java.security.PublicKey;

public class InitialHandshakePacket extends Packet {

    public InitialHandshakePacket(PublicKey publicKey) {
        this.publicKey = RSAEncryptionUtil.toBase64(publicKey);
    }

    private String publicKey;

    public PublicKey getPublicKey() {
        return RSAEncryptionUtil.toPublicKey(publicKey);
    }

}

File: java/core/src/main/java/com/github/fernthedev/lightchat/core/packets/handshake/InitialHandshakePacket.kt
package com.github.fernthedev.lightchat.core.packets.handshake

import com.github.fernthedev.lightchat.core.VersionData
import com.github.fernthedev.lightchat.core.encryption.util.RSAEncryptionUtil
import com.github.fernthedev.lightchat.core.packets.PacketInfo
import com.github.fernthedev.lightchat.core.packets.PacketJSON
import java.security.PublicKey


@PacketInfo(name = "INITIAL_HANDSHAKE_PACKET")
class InitialHandshakePacket private constructor(
    private val publicKey: String,
    private val versionData: VersionDataString
) : PacketJSON() {



    constructor(publicKey: PublicKey, versionData: VersionData) : this(
        RSAEncryptionUtil.toBase64(publicKey),
        VersionDataString(versionData.version.toString(), versionData.minVersion.toString())
    )


    fun getVersionData(): VersionData {
        return VersionData(versionData.version, versionData.minVersion)
    }

    fun getPublicKey(): PublicKey? {
        return RSAEncryptionUtil.toPublicKey(publicKey)
    }
}